x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/transport/CrossClusterAccessServerTransportFilter.java

```diff
@@ -30,7 +30,7 @@
 import static org.elasticsearch.xpack.core.security.authc.CrossClusterAccessSubjectInfo.CROSS_CLUSTER_ACCESS_SUBJECT_INFO_HEADER_KEY;
 import static org.elasticsearch.xpack.security.authc.CrossClusterAccessHeaders.CROSS_CLUSTER_ACCESS_CREDENTIALS_HEADER_KEY;
 
-final class CrossClusterAccessServerTransportFilter extends ServerTransportFilter {
+final class CrossClusterAccessServerTransportFilter extends DefaultServerTransportFilter implements ServerTransportFilter {
```

Comment:
`implements ServerTransportFilter` is implicit since `CrossClusterAccessServerTransportFilter` is a `DefaultServerTransportFilter` that in turn implements `ServerTransportFilter`. 

Might be out of scope for this PR, but the inheritance here is a little confusing. `CrossClusterAccessServerTransportFilter` is only overriding the `authenticate` method which leads me to believe that the code in `CrossClusterAccessServerTransportFilter` actually belongs in `CrossClusterAccessAuthenticationService` and this class could be dropped for that reason.